### PR TITLE
Add a README for the QCheck-STM plugin and amend the main README accordingly

### PR DESCRIPTION
**Pull request by shym**
This:
- adds a README for the new QCheck-STM plugin
- amends the root README with the new plugin and to point to the new README
- fixes a couple of typos in the main README I noticed while amending it.

I wonder about the fact that I’ve naturally written “QCheck-STM” but the documentation of the qcheck-stm package seems to more or less consistently use all lowercase. Browsing the documentation of the qcheck-core package didn’t help me arbitrate: I saw occurrences only of the (lowercase) package name and (capitalised) module name but not the middle ground of plain-English talking about the library. Any opinion, @jmid?
(The `.opam` file should be updated with the chosen spelling, by the way)

### CODE REVIEW

**Comment by shym** [comment]
Side note: we kept for later the description of the generated module as we intend to improve the plugin shortly. The description will be more interesting when this is done.

**Comment by jmid** [comment]
I tried to use lower case `qcheck-stm` and `qcheck-lin` to refer to the opam package names, and upper-case `STM` and `Lin` for the code module names. I still mess up confusing the names of `ocamlfind` and `findlib` so we should be careful about adding needless confusion when the names differ.

File: README.md [hunk 1]
```diff
@@ -140,20 +162,26 @@ module Lib = LibAsserts
 
 ## Supported Gospel
 
-The default mode and the monolith plugin have currently some
-limitations on what Gospel specifications are supported.
+The various modes have currently some limitations on what Gospel
```
Comment (jmid) [inline README.md]: Nit: "The various modes currently have"

File: plugins/qcheck-stm/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+From there, you can run the program directly for random testing:
+
+```shell
+$ ./path/to/main.exe
+```
+
+
+## Supported Gospel and other limitations
+
+The QCheck-STM plugin has currently some limitations on what Gospel
```
Comment (jmid) [inline plugins/qcheck-stm/README.md]: Similar nit: "The QCheck-STM plugin currently has"

**Review by jmid — COMMENTED** [review]
I gave the README changes a look - it looks great!
I commented on a few nits.

**Pull request merged**